extensions/org.wso2.carbon.config.maven.plugin/src/main/java/org/wso2/carbon/config/maven/plugin/ProductConfigurationMojo.java

```diff
@@ -0,0 +1,235 @@
+                    }
+                }
+            } catch (IOException e) {
+                throw new MojoExecutionException("Error while opening feature jarfile", e);
+            } finally {
+                try {
+                    if (outputStream != null) {
+                        outputStream.close();
+                    }
+                    if (inputStream != null) {
+                        inputStream.close();
+                    }
+                } catch (IOException e) {
+                    logger.error("Error while closing the input stream", e);
+                }
+
+            }
+        }
+
+        for (File file : pluginsList) {
+            try (JarFile jar = new JarFile(file)) {
+                JarEntry configDocs = jar.getJarEntry(CONFIG_DOCS_DIR);
+                if (configDocs != null) {
+                    Enumeration enumEntries = jar.entries();
+                    while (enumEntries.hasMoreElements()) {
+                        JarEntry jarEntry = (JarEntry) enumEntries.nextElement();
+                        if (!jarEntry.getName().startsWith("config-docs/")) {
+                            continue;
+                        }
+                        File destFile = new File(outputDirectory.getParent(), jarEntry.getName());
+                        if (jarEntry.isDirectory()) {
+                            if (!destFile.exists() && !destFile.mkdirs()) {
+                                throw new MojoExecutionException("Error while creating config directory in classpath");
+                            }
+                            continue;
+                        }
+                        inputStream = jar.getInputStream(jarEntry);
+                        outputStream = new FileOutputStream(destFile);
+                        while (inputStream.available() > 0) {
+                            outputStream.write(inputStream.read());
+                        }
+                    }
+                }
+            } catch (IOException e) {
+                throw new MojoExecutionException("Error while opening feature jarfile", e);
+            } finally {
+                try {
+                    if (outputStream != null) {
+                        outputStream.close();
+                    }
+                    if (inputStream != null) {
+                        inputStream.close();
+                    }
+                } catch (IOException e) {
+                    logger.error("Error while closing the input stream", e);
+                }
+
+            }
+        }
+
+        Map<String, String> menuMap = new LinkedHashMap<>();
+
+        try {
+            generateIndex(generateMenu(menuMap, new File(outputDirectory.getParent(), CONFIG_DOCS_DIR)));
+        } catch (IOException e) {
+            throw new MojoExecutionException("Error while executing the method", e);
+        }
+    }
+
+    private Map<String, String> generateMenu(Map<String, String> menuMap, File directory)
```

Comment:
Why we need to send empty map as menuMap as an input parameter. We can remove it.